tool/src/js/conversions.rs

```diff
@@ -364,119 +327,62 @@ pub fn gen_value_rust_to_js<W: fmt::Write>(
-            assert!(
-                underlying.is_pointer(),
-                "Options must contain pointer types"
-            );
-
-            writeln!(
-                out,
-                "if (out.{}.underlying !== 0) {if_true} else {if_false}",
-                name,
-                if_true = display::block(|mut f| {
-                    gen_box_destructor(name, underlying.as_ref(), in_path, env, &mut f)
-                }),
-                if_false = display::block(|mut f| {
-                    writeln!(
-                        f,
-                        "Object.defineProperty(out, \"{}\", {{ value: null }});",
-                        name
-                    )
-                })
-            )?;
-            // TODO(#62): don't generate destructor if null
-        }
-
-        _ => {}
-    }
-
-    Ok(())
-}
-
-/// TODO: docs
-fn gen_rust_reference_to_js<W: fmt::Write>(
+pub fn gen_rust_reference_to_js<W: fmt::Write>(
     underlying: &ast::TypeName,
     in_path: &ast::Path,
     value_expr: &str,
-    owner: &str,
     borrowed_params: &ast::BorrowedParams,
     env: &Env,
     out: &mut W,
 ) -> fmt::Result {
     match underlying {
-        ast::TypeName::Box(typ) => {
-            gen_rust_reference_to_js(
-                typ.as_ref(),
-                in_path,
-                &format!(
-                    "(new Uint32Array(wasm.memory.buffer, {}, 1))[0]",
-                    value_expr
-                ),
-                owner,
-                borrowed_params,
-                env,
-                out,
-            )?;
-        }
-
-        ast::TypeName::Reference(.., typ) => {
+        ast::TypeName::Box(typ) | ast::TypeName::Reference(.., typ) => {
             gen_rust_reference_to_js(
                 typ.as_ref(),
                 in_path,
                 &format!(
                     "(new Uint32Array(wasm.memory.buffer, {}, 1))[0]",
                     value_expr
                 ),
-                "null", // TODO(#12): pass in lifetime of the reference
                 borrowed_params,
                 env,
                 out,
             )?;
         }
-
         ast::TypeName::Option(underlying) => match underlying.as_ref() {
-            ast::TypeName::Box(_) => {
-                // TODO(#62): return null if pointer is 0
-                gen_rust_reference_to_js(
-                    underlying.as_ref(),
-                    in_path,
-                    value_expr,
-                    owner,
-                    borrowed_params,
-                    env,
-                    out,
-                )?;
+            ast::TypeName::Box(..) | ast::TypeName::Reference(..) => write!(
+                out,
+                "{}",
+                display::iife(|mut f| {
+                    let ident = ast::Ident::from("option_ptr");
+                    writeln!(
+                        f,
+                        "const {ident} = (new Uint32Array(wasm.memory.buffer, {}, 1))[0];",
```

Comment:
this has the effect of reading the data _behind_ the pointer, which is incorrect